### DISCUSSION TITLE
Menu: Allow custom templates for headers

### DISCUSSION THREAD

**Discussion by thetric** [primefaces/community#508] [PrimeNG] [answered]
Like for regular items, it should be possible to define custom templates for headers in menus. See https://github.com/primefaces/primeng/blob/master/src/app/components/menu/menu.ts#L185

Background: I want to use templates to move the translation logic there instead of the component code (subscribe for language changes):

```html
    <p-menu [model]="navigationStructure">
      <ng-template pTemplate="item" let-item>
        <a class="p-menuitem-link flex justify-content-between align-items-center p-3">
          {{ item.label | translate }}  <!-- THIS -->
        </a>
      </ng-template>
    </p-menu>
```

**Comment by cetincakiroglu** [comment, marked as answer]
Thanks for the feedback, we've added it in our internal backlog.
